### DISCUSSION TITLE
Is there an EBNF for D2?

### DISCUSSION THREAD

**Discussion by grosenberg** [terrastruct/d2#1174] [General]
Looking for a D2 EBNF (or similar) language description.  

This would be the most efficient starting point for me to add D2 support to [Fluentmark](https://github.com/grosenberg/Fluentmark), an Eclipse-based markdown editor.

Thanks

**Comment by alixander** [comment]
there isn't yet unfortunately, the closest is our regex syntax highlighters.

vscode: https://github.com/terrastruct/d2-vscode/blob/master/syntaxes/d2.tmLanguage.yaml
vim: https://github.com/terrastruct/d2-vim/blob/master/syntax/d2.vim
